docs/tools/apiExtractorUtils.ts

```diff
@@ -0,0 +1,385 @@
+import {
+  ApiItem,
+  ApiModel,
+  ApiParameterListMixin,
+  ApiDocumentedItem,
+  ApiPackage,
+  ApiPropertyItem,
+  Excerpt,
+  ApiMethod,
+  ApiReturnTypeMixin,
+  ApiReleaseTagMixin,
+  ReleaseTag,
+  ApiDeclaredItem,
+} from '@microsoft/api-extractor-model';
+import * as h from './html';
+import { toMarkdownString, toHtmlString } from './stringify';
+
+import Fs from 'fs/promises';
+import { readFileSync } from 'fs';
+
+import Prettier from 'prettier';
+
+import {
+  root,
+  paragraph,
+  text,
+  heading,
+  list,
+  listItem,
+  brk,
+  rootWithTitle,
+  Children,
+  link,
+  code,
+  inlineCode,
+  table,
+  tableRow,
+  tableCell,
+  html,
+  strong,
+  emphasis,
+} from 'mdast-builder';
+import { Node, Parent, Literal } from 'unist';
+
+import Path from 'path';
+
+import { ApiItemKind, Context } from './types';
+import { getTSBlock, getTSBlockInHtml, isEmptyOrWhitespace, printTsDoc } from './tsdoc';
+import { rimraf, getFilePath, getHashLink, referenceToLink, getFileUrl } from './files';
+import { StandardTags } from '@microsoft/tsdoc';
+
+const prettierrc = JSON.parse(readFileSync(Path.join(process.cwd(), '..', '.prettierrc'), 'utf-8'));
+
+type GroupName = keyof typeof GROUPS;
+const GROUPS = {
+  Types: [ApiItemKind.Interface, ApiItemKind.TypeAlias],
+  Enums: [ApiItemKind.Enum],
+  Functions: [ApiItemKind.Function],
+  Classes: [ApiItemKind.Class],
+  Other: [],
+} as const;
+
+const API_ITEM_KIND_TO_GROUP = (Object.entries(GROUPS) as Array<[GroupName, readonly ApiItemKind[]]>).reduce<
+  Partial<Record<ApiItemKind, GroupName>>
+>((acc, [group, kinds]) => {
+  kinds.forEach((kind) => (acc[kind] = group));
+  return acc;
+}, {});
+
+async function run() {
+  const apiModel = new ApiModel();
+  const apiContainer = apiModel.loadPackage(Path.join('server.api.json'));
+
+  await rimraf();
+  return printIndexPage(apiModel, apiContainer);
+}
+
+async function save(apiItem: ApiItem, content: Parent) {
+  const filePath = getFilePath(apiItem);
+  const markdown = toMarkdownString(content);
+  await Fs.writeFile(filePath, Prettier.format(markdown, { ...prettierrc, parser: 'markdown' }));
+}
+
+async function printIndexPage(apiModel: ApiModel, apiItem: ApiPackage) {
```

Comment:
These names are quite generic to me, as I'm not that experienced in this type of libraries, I guess that `run` & `save` are okay after reading their content. Why did you decide to name all these functions `print` instead of `generate` or something like that. Is it a convention?

Reply:
No idea